### DISCUSSION TITLE
Wait for stream

### DISCUSSION THREAD

**Discussion by ysnist** [bluenviron/mediamtx#1413] [General]
I have a video archive and ffmpeg sends the movies in my video archive to my rtmp server in order, but each time I switch to a new movie, the server connection is momentarily interrupted and the stream is naturally restarted on my rtmp server. Can I prevent my users from disconnecting until the new stream is sent when my rtmp server is interrupted for a few seconds?

**Comment by aler9** [comment]
Hello, this is not possible with RTMP, since a new RTMP publisher can publish different tracks or codecs from the previous one, and tracks / codecs are sent once to readers, during the handshake with server, therefore readers couldn't read the new stream anyway.

Use WebRTC or HLS, since they support publisher reconnection.

**Comment by ysnist** [comment]
I realized that it was possible to do this in SRS and I started using it, no problem at all.
Smooth transition if the broadcaster sends new video with the same codec and same fps after terminating the connection

https://github.com/ossrs/srs

[a comment by an automated account is not shown]
